### ISSUE TITLE
Invalid ignore (or select) arguments silently ignored

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#580]
In GitLab by @pjacock on May 1, 2017, 07:01

*Please describe how you installed Flake8*

Example:

```
$ pip3 install flake8 flake8-docstring
```

*Please provide the exact, unmodified output of `flake8 --bug-report`*

```
$ flake8 --bug-report
{
  "dependencies": [
    {
      "dependency": "setuptools",
      "version": "28.8.0"
    }
  ],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.6.0",
    "system": "Darwin"
  },
  "plugins": [
    {
      "plugin": "flake8-docstrings",
      "version": "1.1.0, pydocstyle: 2.0.0"
    },
    {
      "plugin": "flake8_commas._base",
      "version": "0.4.3"
    },
    {
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.3.1"
    },
    {
      "plugin": "pyflakes",
      "version": "1.5.0"
    }
  ],
  "version": "3.3.0"
}
```

*Please describe the problem or feature*

Accidental configuration errors are silently ignored.

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

Sample test file:

```
$ python3 nasty.py 
42
$ python nasty.py 
42
$ cat nasty.py 
"""Test case for flake8 with some deliberate bad style."""


#bad spacing, and docstring issues too
def add(x,y):
    ''' this is a pointless function which will add two variables together
    withouth making any assumptions about their type '''
    return x+y


print(add(40, 2))
```

Running flake8 in isolated mode to show all possible warnings including E226:

```
$ flake8 nasty.py --isolated --ignore ""
nasty.py:4:1: E265 block comment should start with '# '
nasty.py:5:1: D205 1 blank line required between summary line and description
nasty.py:5:1: D209 Multi-line docstring closing quotes should be on a separate line
nasty.py:5:1: D210 No whitespaces allowed surrounding docstring text
nasty.py:5:1: D300 Use """triple double quotes"""
nasty.py:5:1: D400 First line should end with a period
nasty.py:5:1: D401 First line should be in imperative mood; try rephrasing
nasty.py:5:1: D403 First word of the first line should be properly capitalized
nasty.py:5:10: E231 missing whitespace after ','
nasty.py:8:13: E226 missing whitespace around arithmetic operator
```

Running flake8 with a well formed ignore list:

```
$ flake8 nasty.py --isolated --ignore "E265,D2,D4"
nasty.py:5:1: D300 Use """triple double quotes"""
nasty.py:5:10: E231 missing whitespace after ','
nasty.py:8:13: E226 missing whitespace around arithmetic operator
```

Running flake8 with a malformed ignore list:

```
$ flake8 nasty.py --isolated --ignore "E265,D2 D4"
nasty.py:5:1: D205 1 blank line required between summary line and description
nasty.py:5:1: D209 Multi-line docstring closing quotes should be on a separate line
nasty.py:5:1: D210 No whitespaces allowed surrounding docstring text
nasty.py:5:1: D300 Use """triple double quotes"""
nasty.py:5:1: D400 First line should end with a period
nasty.py:5:1: D401 First line should be in imperative mood; try rephrasing
nasty.py:5:1: D403 First word of the first line should be properly capitalized
nasty.py:5:10: E231 missing whitespace after ','
nasty.py:8:13: E226 missing whitespace around arithmetic operator
```

Expected result here would be an error about the ignore list.

Actual result was as for:

```
$ flake8 nasty.py --isolated --ignore "E265"
nasty.py:5:1: D205 1 blank line required between summary line and description
nasty.py:5:1: D209 Multi-line docstring closing quotes should be on a separate line
nasty.py:5:1: D210 No whitespaces allowed surrounding docstring text
nasty.py:5:1: D300 Use """triple double quotes"""
nasty.py:5:1: D400 First line should end with a period
nasty.py:5:1: D401 First line should be in imperative mood; try rephrasing
nasty.py:5:1: D403 First word of the first line should be properly capitalized
nasty.py:5:10: E231 missing whitespace after ','
nasty.py:8:13: E226 missing whitespace around arithmetic operator
```

The command line version is annoying, but the configuration file case is more easily triggered:

```
$ cat flake8_good 
[flake8]
ignore=
    E265,
    D2,
    D4
$ flake8 nasty.py --config flake8_good 
nasty.py:5:1: D300 Use """triple double quotes"""
nasty.py:5:10: E231 missing whitespace after ','
nasty.py:8:13: E226 missing whitespace around arithmetic operator
```

vs:

```
$ cat flake8_bad 
[flake8]
ignore=
    E265,
    D2
    D4
$ flake8 nasty.py --config flake8_bad
nasty.py:5:1: D205 1 blank line required between summary line and description
nasty.py:5:1: D209 Multi-line docstring closing quotes should be on a separate line
nasty.py:5:1: D210 No whitespaces allowed surrounding docstring text
nasty.py:5:1: D300 Use """triple double quotes"""
nasty.py:5:1: D400 First line should end with a period
nasty.py:5:1: D401 First line should be in imperative mood; try rephrasing
nasty.py:5:1: D403 First word of the first line should be properly capitalized
nasty.py:5:10: E231 missing whitespace after ','
nasty.py:8:13: E226 missing whitespace around arithmetic operator
```

Here as per http://flake8.pycqa.org/en/latest/user/configuration.html#configuration I am deliberately using multiple lines for the ignore list (and would actually have comments in here too).

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on May 12, 2017, 17:01

So, I'm not entirely certain that we can error on that. We don't presently restrict error codes from having `\n`s in them, and while I doubt anyone is taking advantage of that, I'm not sure that we want to generate an error in this case. Perhaps, we should just Do The Right Thing?

This would mean expanding how we accept `--select` and `--ignore` such that people can do:

```
$ flake8 --select "E123 E234 E345" --ignore "D123 D234 D345"
$ flake8 --select "E1,E2, E3 E4"
```

And

```
[flake8]
ignore =
    E2
    D2
    D4
```

In addition to what we already do? How does that sound to you?

**Comment by asottile** [comment]
In GitLab by @pjacock on May 13, 2017, 01:55

I think splitting on white space as well as commas is a good idea for usability, cross reference #330 which can likely be fixed at the same time.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on May 25, 2017, 18:21

mentioned in merge request !185

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on May 25, 2017, 18:27

closed via merge request !185

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on May 25, 2017, 18:27

closed via commit a42299d008df7054532d52f84b0795d0e76cf5a8

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on May 25, 2017, 18:27

closed via commit d890b8b6833689aa0a67abb8f406082f4dd8ad81

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on May 27, 2017, 11:56

mentioned in commit jdufresne/flake8@d890b8b6833689aa0a67abb8f406082f4dd8ad81
